### ISSUE TITLE
matter 1.0 branch pairing failed with sve2(v2.5)TH

### ISSUE THREAD

**Issue by Leo2442926161** [project-chip/connectedhomeip#23218]
### Reproduction steps

```bash
DUT: matter 1.0 commit: 26a054c31144478e3c27391c06b0dc3230bb0882 (HEAD -> v1.0-branch, origin/v1.0-branch)
TH  : TH v 2.5 - 1.0 Release Build

The CD in sve2 and matter 1.0 is not same, the links are below, I am wondering whether the cd or keys are mismatched,we  should updated the TH image.
https://github.com/project-chip/connectedhomeip/blob/sve-2/src/credentials/examples/DeviceAttestationCredsExample.cpp
https://github.com/project-chip/connectedhomeip/blob/v1.0-branch/src/credentials/examples/DeviceAttestationCredsExample.cpp
```

### Bug prevalence

everytime

### GitHub hash of the SDK that was being used

26a054c31144478e3c27391c06b0dc3230bb0882 

### Platform

efr32, raspi

### Platform Version(s)

ubuntu 22.04

### Type

Core SDK Interopability Issue

### Anything else?

The chip tool on the TH log says **"Failed in verifying 'Attestation Information' command received from the device: err 601. Look at AttestationVerificationResult enum to understand the errors"**  and I checked the enum(**kCertificationDeclarationNoCertificateFound = 601**) means no certificate found, but it works very well with  matter1.0-branch chip tool, dose any one can help me out? thank you very much.
[pairing failed.txt](https://github.com/project-chip/connectedhomeip/files/9797999/pairing.failed.txt)

**Comment by Leo2442926161** [comment]
![image](https://user-images.githubusercontent.com/30258768/196199189-04e11a8e-3b9f-415e-b7d4-9e17d6106c0f.png)

**Comment by tcarmelveilleux** [comment]
In the above comment:

Left side is 1.0 branch (https://github.com/project-chip/connectedhomeip/blob/v1.0-branch/src/credentials/examples/DeviceAttestationCredsExample.cpp), right side is master (https://github.com/project-chip/connectedhomeip/blob/master/src/credentials/examples/DeviceAttestationCredsExample.cpp#L82)

**Comment by emargolis** [comment]
@Leo2442926161  the appropriate fixes were cherry-picked with #23239 

Please test

**Comment by Aimeezhang2022** [comment]
> @Leo2442926161 the appropriate fixes were cherry-picked with #23239
> 
> Please test

Hi @emargolis , based on the TH2.5 (SHA: 6e5a2f9) , what should I do to fix this in TH? Thank you.
